### PR TITLE
Custom `EthTxManager` on `agglayer`

### PR DESCRIPTION
**Pull request by goran-ethernal**
This PR copies `ethTxManager` implementation from `zkevm-node` to `agglayer` and implements custom logic the `agglayer` needs.

Since `zkevm` team does not want changes done to the original `ethTxManager`, the goal of this PR is to copy all the logic from it (including the fix from PR: https://github.com/0xPolygonHermez/zkevm-node/pull/3338 regarding the nonce hole issue), and add an extra feature of dropping monitored transactions that fail for a configured number of times. It also removed all the functions that `agglayer` does not use from `ethTxManager`, like `Reorg`.
We also needed to copy `pgstorage.go` since we can not use the one from `zkevm-node` since it uses the non-exposed `monitoredTx` struct.

### Issue that we faced

During testing of the agglayer on testnet, an issue popped up where a transaction was repeated indefinitely even though for some reason, it constantly failed to be executed or it failed to estimate gas. This caused unnecessary traffic, load and logs, since that transaction was never going to be executed successfully, since it failed, 100s of times.

### Fix
This PR introduces a new configuration parameter to the `EthTxManager` configuration on `agglayer` called `MaxRetries`. It represents a maximum number a transaction will be resent to the L1, or a maximum number of times it will try to estimate its gas until it drops it (updates the monitored transaction with the `MonitoredTxStatusFailed` status in the `agglayer-db`), meaning, it will not try to resend it again.

In order to know how many times a transaction was tried to be resent or estimated, a new column to the `state.monitored_txs` table needed to be added, called `num_retries`. That is why, a new migration file was added `0002.sql` which alters given table, by adding the new column to it.

### Tests
This PR copied the original `ethTxManager` tests, as well as tests for its postgress storage. Tests added:
- `monitoredtx_test.go` - this is the same as in original `ethTxManager`, no need to review it.
- `pgstorage_test.go` -  this is the same as in original `ethTxManager`, no need to review it.
- `txmanager_test.go` - this is the same as in original `ethTxManager`. Only need to review test `TestTxRetry_MaxRetriesReached` which tests the new logic.

### How to review this PR
In order not to waste time reviewing the copy/pasted code this is what should be considered for reviewing:
- `main.go`
- `config.go`
- `default.go`
- `0002.sql`
- `agglayer.toml`
- `executor_test.go`
- `executor.go`
- `rpc_test.go`
- `pgstorage.go `- only changes regarding the `num_retries` column.
- `txmanager.go` - parts of code where `mtx.NumRetries` and `cfg.MaxRetries` are mentioned. Other parts of code were left the same.
- `txmanager_test.go` - only test `TestTxRetry_MaxRetriesReached`, other tests are just moved from original `ethTxManager`.
-`types/interfaces.go`
- `Make file`

### CODE REVIEW

File: db/migrations/0002.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-- +migrate Up
```
[the remark this answers is not in the document]
Comment (goran-ethernal) [reply db/migrations/0002.sql:1]: Fix: https://github.com/0xPolygon/agglayer/pull/109/commits/4efe478f3a5ddeb041ca11816e79d780f0e4f21b

File: txmanager/pgstorage.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+	mTxs := []txmTypes.MonitoredTx{}
+
+	var rows pgx.Rows
+	var err error
+	if hasStatusToFilter {
+		rows, err = conn.Query(ctx, cmd, owner, statuses)
+	} else {
+		rows, err = conn.Query(ctx, cmd, owner)
+	}
+
```
Comment (christophercampbell) [inline txmanager/pgstorage.go:108]: I realize you copied this code from elsewhere, but there should be a `defer rows.Close()` here

File: txmanager/pgstorage.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	mTxs := []txmTypes.MonitoredTx{}
+
+	var rows pgx.Rows
+	var err error
+	if hasStatusToFilter {
+		rows, err = conn.Query(ctx, cmd, sender.String(), statuses)
+	} else {
+		rows, err = conn.Query(ctx, cmd, sender.String())
+	}
+
```
Comment (christophercampbell) [inline txmanager/pgstorage.go:154]: `defer rows.Close()`

File: txmanager/pgstorage.go (hunk 2 above)
Comment (goran-ethernal) [reply txmanager/pgstorage.go:108]: Indeed. Good catch. Roman actually added this through another PR on the zkevm node. I'll add it here as well.

File: txmanager/pgstorage.go (hunk 3 above)
Comment (goran-ethernal) [reply txmanager/pgstorage.go:156]: Fix: https://github.com/0xPolygon/agglayer/pull/109/commits/6e0a701826e5532bcd6214bb28c833473b5c2d4f

**Review by rachit77 — APPROVED** [review]

**Pull request merged**